metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -831,7 +834,9 @@ abstract class MetalsLspService(
         val path = params.getTextDocument.getUri.toAbsolutePath
         buffers.put(path, change.getText)
         diagnostics.didChange(path)
-        compilers.didChange(path)
+        compilers
+          .didChange(path)
+          .map(diagnostics.publishDiagnosticsNotAdjusted(path, _))
```

Comment:
Diagnostics coming from the PC are already correctly positioned relative to in-memory buffers. Not sure if there is a cleaner way than this.